### DISCUSSION TITLE
Getting started using barseip as library

### DISCUSSION THREAD

**Discussion by gbernaldo82** [baresip/baresip#2272] [Q&A]
Hi, good afternoon!

I'm getting started using barseip as a library for one of my projects and I wonder if there is any kind of API or Documentation about the source code, I mean, something more than the source code. I tried to look for some examples on google but I couldn't find anything related. I just want to learn to use the library but I don't want to get into the source code if not needed. 

Right now, the only thing I need is to make a call to one of my Asterisk servers and I want to do it in C/C++. 

Please, can anyone help me with that? I don't know exactly where to begin.
Thank you in advance!

**Comment by sreimers** [comment]
Here is a good starting point: https://github.com/baresip/baresip/wiki/Using-baresip-as-a-library

**Reply by gbernaldo82** [reply]
Thank you so much!

I already saw the wiki, but I need some thing more like an API (maybe some kind of documentation made with Doxygen or something like that)

Thanks again!

**Reply by sreimers** [reply]
You can checkout the project and compile doxygen docs with `make dox`

**Reply by gbernaldo82** [reply]
Many thanks @sreimers 
Let me check ir out!
